packages/frontend/app/services/channels/vote-handler.ts

```diff
@@ -0,0 +1,130 @@
+import Service, { inject as service } from '@ember/service';
+import Message from 'emberclear/models/message';
+import StoreService from '@ember-data/store';
+import VoteVerifier from './vote-verifier';
+import FindOrCreateChannelService from './find-or-create';
+import VoteChain, { VOTE_ACTION } from 'emberclear/models/vote-chain';
+import { identityEquals } from 'emberclear/utils/identity-comparison';
+import Channel from 'emberclear/models/channel';
+import { isVoteCompleted, isVoteCompletedPositive } from './-utils/vote-completion';
+import ChannelContextChain from 'emberclear/models/channel-context-chain';
+import Vote from 'emberclear/models/vote';
+
+export default class ReceivedChannelVoteHandler extends Service {
+  @service store!: StoreService;
+  @service('channels/vote-verifier') voteVerifier!: VoteVerifier;
+  @service('channels/find-or-create') findOrCreator!: FindOrCreateChannelService;
+
+  public async handleChannelVote(message: Message, raw: StandardMessage) {
+    let existingChannel = await this.findOrCreator.findOrCreateChannel(raw.channelInfo);
+    if (existingChannel?.members.contains(message.sender!)) {
+      let sentVote = message.metadata as StandardVote;
+      let existingVote = await this.findOrCreator.findOrCreateVote(sentVote);
+
+      if (existingVote !== undefined) {
+        let voteChain = await this.findOrCreator.findOrCreateVoteChain(sentVote.voteChain);
```

Comment:
Should this be a findOrCreate? In theory, incoming votes should never be found in our system and may cause issues below.